src/DynamoPublish/ViewModels/PublishViewModel.cs

```diff
@@ -0,0 +1,103 @@
+﻿using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Publish.Models;
+using Dynamo.UI.Commands;
+using Dynamo.ViewModels;
+using Dynamo.Wpf.Authentication;
+using System;
+using System.Collections.Generic;
+using System.Windows.Forms;
+using System.Windows.Input;
+
+namespace Dynamo.Publish.ViewModels
+{
+    public class PublishViewModel : NotificationObject
+    {
+        #region Properties
+
+        private string name;
+        public string Name
+        {
+            get { return name; }
+            set
+            {
+                name = value;
+                RaisePropertyChanged("Name");
+            }
+        }
+
+        private string description;
+        public string Description
+        {
+            get { return description; }
+            set
+            {
+                description = value;
+                RaisePropertyChanged("Description");
+            }
+        }
+
+        private string shareLink;
+        public string ShareLink
+        {
+            get { return shareLink; }
+            set
+            {
+                shareLink = value;
+                RaisePropertyChanged("ShareLink");
+            }
+        }
+
+        private PublishModel model;
+        public PublishModel Model
+        {
+            get { return model; }
+        }
+
+        /// <summary>
+        /// Used as parent window for authentication window.
+        /// </summary>
+        public Views.PublishView PublishView { get; set; }
+
+        public IEnumerable<IWorkspaceModel> WorkSpaces { get; set; }
+
+        #endregion
+
+        #region Click commands
+
+        public ICommand PublishCommand { get; private set; }
+
+        #endregion
+
+        #region Initialization
+
+        public PublishViewModel(PublishModel model)
+        {
+            this.model = model;
+
+            PublishCommand = new DelegateCommand(OnPublish);
+        }
+
+        #endregion
+
+
+        #region Helpers
+
+        private void OnPublish(object obj)
+        {
+            if (!model.IsLoggedIn)
+            {
+                model.LoginService = new LoginService(PublishView, new WindowsFormsSynchronizationContext());
```

Comment:
Let's create an event `OnRequestLogin`.  `PublishView` can subscribe.  When this event is emitted `PublishView` can supply itself as the parent window in the callback like:

`LoginService( this, ...`